apply-roles

```diff
@@ -133,24 +133,29 @@ main() {
     exit 1
   fi
 
-  # This will set a global variable if the custom role has been created
-  check_for_custom_role_availability
-
   # retrieving Tenant id for Active Directory
   tenant_id=$(az ad sp show --id "${application_id}" --query "appOwnerTenantId" --output tsv)
 
+  # ensure list of subscriptions is up to date with the server
+  az account list --refresh > /dev/null
+
   # retrieve list of all subscriptions in the Active Directory and convert into a bash array
   subscription_ids_in_active_directory=()
   # Bash v3 supported method of converting to an array ref: https://github.com/koalaman/shellcheck/wiki/SC2207
   while IFS='' read -r line; do
     subscription_ids_in_active_directory+=("$line");
   done < <(az account list --query "[?tenantId=='${tenant_id}'].id" --output tsv)
 
+  echo "Configuring custom \"${CUSTOM_ROLE_NAME}\" role definition:"
+
+  # This will set a global variable if the custom role has been created
+  check_for_custom_role_availability
```

Comment:
Had to move this here so the logging was displayed nicely, though it is a more logical place for this call to be made anyway